src/asm/section.cpp

```diff
@@ -862,71 +932,65 @@ void sect_BinaryFileSlice(std::string const &name, int32_t startPos, int32_t len
 		error("Number of bytes to read cannot be negative (%" PRId32 ")\n", length);
 		length = 0;
 	}
-	if (!checkcodesection())
+	uint8_t *bytes = reserveBytes(length);
+	if (bytes == nullptr)
 		return;
+	growSection(length); // We are able to immediately grow the section accordingly.
 	if (length == 0) // Don't even bother with 0-byte slices
 		return;
 
-	FILE *file = nullptr;
+	int file = -1;
 	if (std::optional<std::string> fullPath = fstk_FindFile(name); fullPath)
-		file = fopen(fullPath->c_str(), "rb");
-	if (!file) {
+		file = open(fullPath->c_str(), O_RDONLY | O_BINARY);
```

Comment:
The incbin logic is getting complex enough that it would be nice to factor it out into something reusable for both slices and whole files, unless that would be too unavoidably inefficient. (Honestly it was already at that point.)